trace/trace.go

```diff
@@ -504,3 +504,14 @@ func (s *Span) createChildSpan(ctx context.Context, async bool) (context.Context
 	ctx = PutSpanInContext(ctx, newSpan)
 	return ctx, newSpan
 }
+
+// PropagationContext creates and returns a new propagation.PropagationContext using the
+// information in the current span.
+func (s *Span) PropagationContext() *propagation.PropagationContext {
+	return &propagation.PropagationContext{
+		TraceID:      s.trace.traceID,
+		ParentID:     s.spanID,
+		Dataset:      s.trace.builder.Dataset,
+		TraceContext: s.trace.traceLevelFields,
+	}
+}
```

Comment:
went from "why is this here?" to "right, because _everybody wants access to the propagation context_ and is seriously annoyed when the only format available is the serialized format. 👍